lib/quantum/application.ex

```diff
@@ -0,0 +1,20 @@
+defmodule Quantum.Application do
+  @moduledoc false
+
+  use Application
+
+  @dependency_application Application.get_env(:quantum, :date_library, Quantum.DateLibrary.Timex)
+    .dependency_application()
+
+  case @dependency_application do
```

Comment:
Ah my bad, that needs to be `Application.ensure_all_started(unquote(deps), :permanent)` because that `if` is still in `defmodule` context...
Either way... 👍